src/include/detail/linalg/tdb_io.h

```diff
@@ -43,6 +43,77 @@
 #include "utils/print_types.h"
 #include "utils/timer.h"
 
+namespace {
+
+template <class T>
+std::vector<T> read_vector_helper(
+    const tiledb::Context& ctx,
+    const std::string& uri,
+    size_t start_pos,
+    size_t end_pos,
+    size_t timestamp,
+    bool read_full_vector) {
```

Comment:
Rather than having a flag, we should just omit `start_pos` and `end_pos` if we want to read the whole thing.

Reply:
So this is just a helper function in an anonymous namespace, the actual user-facing APIs are:
```
template <class T>
std::vector<T> read_vector(
    const tiledb::Context& ctx,
    const std::string& uri,
    size_t start_pos,
    size_t end_pos,
    size_t timestamp = 0) {
  return read_vector_helper<T>(ctx, uri, start_pos, end_pos, timestamp, false);

template <class T>
std::vector<T> read_vector(
    const tiledb::Context& ctx, const std::string& uri, size_t timestamp = 0) {
  return read_vector_helper<T>(ctx, uri, 0, 0, timestamp, true);
}
```
Do you still want to change it? This approach doesn't seem that bad to me.

Reply:
Okay sounds good, moved this code over to a new PR as-is.